components/inputs/input-search.js

```diff
@@ -0,0 +1,239 @@
+					--d2l-button-icon-border-radius: 4px;
+					--d2l-button-icon-focus-box-shadow: 0 0 0 1px #ffffff, 0 0 0 3px var(--d2l-color-celestine);
+					position: absolute;
+					right: 0.3rem;
+					top: 50%;
+					transform: translateY(-50%);
+				}
+				:host([dir="rtl"]) d2l-button-icon {
+					left: 0.3rem;
+					right: auto;
+				}
+			`
+		];
+	}
+
+	static get resources() {
+		return {
+			'ar': { 'search': 'بحث', 'search.clear': 'مسح البحث' },
+			'en': { 'search': 'Search', 'search.clear': 'Clear Search' },
+			'es': { 'search': 'Buscar', 'search.clear': 'Borrar búsqueda' },
+			'fr': { 'search': 'Rechercher', 'search.clear': 'Effacer la recherche' },
+			'ja': { 'search': '検索', 'search.clear': '検索のクリア' },
+			'ko': { 'search': '검색', 'search.clear': '검색 지우기' },
+			'nl': { 'search': 'Zoeken', 'search.clear': 'Zoekopdracht wissen' },
+			'pt': { 'search': 'Pesquisar', 'search.clear': 'Limpar Pesquisa' },
+			'sv': { 'search': 'Sökning', 'search.clear': 'Rensa sökning' },
+			'tr': { 'search': 'Ara', 'search.clear': 'Aramayı Temizle' },
+			'zh': { 'search': '搜索', 'search.clear': '清除搜索' },
+			'zh-tw': { 'search': '搜尋', 'search.clear': '清除搜尋' }
+		};
+	}
+
+	constructor() {
+		super();
+		this._focussed = false;
+		this._hovered = false;
+		this._lastSearchValue = '';
+		this.disabled = false;
+		this.noClear = false;
+		this.value = '';
+	}
+
+	get lastSearchValue() { return this._lastSearchValue; }
+	set lastSearchValue(val) {}
+	_setLastSearchValue(val) {
+		const oldVal = this._lastSearchValue;
+		this._lastSearchValue = val;
+		this.requestUpdate('lastSearchValue', oldVal);
+	}
+
+	render() {
+		requestAnimationFrame(
+			() => this.dispatchEvent(new CustomEvent('d2l-input-search-render', {
+				bubbles: false,
+				composed: false
+			}))
+		);
+		const inputClasses = {
+			'd2l-input': true,
+			'd2l-input-focus': !this.disabled && (this._focussed || this._hovered)
+		};
+		const showSearch = this._computeShowSearch();
+		return html`
+			<div class="d2l-input-search-container"
+				@mouseout="${this._handleMouseLeave}"
+				@mouseover="${this._handleMouseEnter}">
+				<input
+					aria-label="${ifDefined(this.label)}"
+					class="${classMap(inputClasses)}"
+					?disabled="${this.disabled}"
+					@input="${this._handleInput}"
+					@keypress="${this._handleInputKeyPress}"
+					maxlength="${ifDefined(this.maxlength)}"
+					placeholder="${ifDefined(this.placeholder)}"
+					type="search"
+					.value="${this.value}">
+				<d2l-button-icon
+					class="d2l-input-search-search"
+					?disabled="${this.disabled}"
+					?hidden="${!showSearch}"
+					icon="tier1:search"
+					@click="${this.search}"
+					text="${this.localize('search')}"></d2l-button-icon>
+				<d2l-button-icon
+					class="d2l-input-search-clear"
+					@click="${this._handleClearClick}"
+					?disabled="${this.disabled}"
+					?hidden="${showSearch}"
+					icon="tier1:close-default"
+					text="${this.localize('search.clear')}"></d2l-button-icon>
```

Comment:
Does it make sense to just use `clear` as the term?

Reply:
Indeed -- updated!